test/e2e_node/lifecyclehooks/prestop-hook.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecyclehooks
+
+import (
+    "context"
+    "fmt"
+
+    v1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    "k8s.io/kubernetes/test/e2e/framework"
+    "k8s.io/kubernetes/test/e2e/framework/ssh"
+    e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+    "k8s.io/kubernetes/test/e2e_node/testdoc"
+    admissionapi "k8s.io/pod-security-admission/api"
+    imageutils "k8s.io/kubernetes/test/utils/image"
+    "github.com/onsi/ginkgo/v2"
+    "github.com/onsi/gomega"
+)
+
+var _ = ginkgo.Describe("PreStop Hook Test Suite", func() {
+    f := framework.NewDefaultFramework("prestop-hook-test-suite")
+    f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+    f.It("should run prestop basic with the grace period", f.WithNodeConformance(), func(ctx context.Context) {
+        testdoc.TestName("Hooks:PreStop Hook Basic Test")
+
+        const gracePeriod int64 = 30
+        client := e2epod.NewPodClient(f)
+        volumeName := "host-data"
+        hostPath := "/tmp/prestop-hook-test.log"
+        gracePeriodValue := gracePeriod
+
+        ginkgo.By("creating a pod with a termination grace period and a long-running PreStop hook")
+        pod := &v1.Pod{
+            ObjectMeta: metav1.ObjectMeta{
+                Name: "pod-termination-grace-period",
+            },
+            Spec: v1.PodSpec{
+                Containers: []v1.Container{
+                    {
+                        Name:  "busybox",
+                        Image: imageutils.GetE2EImage(imageutils.BusyBox),
+                        Command: []string{
+                            "sleep",
+                            "10000",
+                        },
+                    },
+                },
+                TerminationGracePeriodSeconds: &gracePeriodValue,
```

Comment:
nit: use `ptr.To(gracePeriod)` instead.

Actually, you can just inline the graceperiod value with `ptr.To[int64](30)`